apis/gpio/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+#![no_std]
+
+use core::marker::PhantomData;
+
+use libtock_platform::{
+    share::Handle, subscribe::OneId, DefaultConfig, ErrorCode, Subscribe, Syscalls, Upcall,
+};
+
+/// The Gpio driver
+///
+/// # Example
+/// ```ignore
+/// use libtock2::Gpios;
+///
+/// // Turn on led 0
+/// let pin = Gpios::get_pin(0)?;
+///
+/// ```
+
+const DRIVER_NUM: u32 = 4;
+
+// Command IDs
+const GPIO_COUNT: u32 = 0;
+
+const GPIO_ENABLE_OUTPUT: u32 = 1;
+const GPIO_SET: u32 = 2;
+const GPIO_CLEAR: u32 = 3;
+const GPIO_TOGGLE: u32 = 4;
+
+const GPIO_ENABLE_INPUT: u32 = 5;
+const GPIO_READ_INPUT: u32 = 6;
+
+const GPIO_ENABLE_INTERRUPTS: u32 = 7;
+const GPIO_DISABLE_INTERRUPTS: u32 = 8;
+
+const GPIO_DISABLE: u32 = 9;
+
+#[derive(Copy, Clone, PartialEq)]
+#[cfg_attr(test, derive(Debug))]
```

Comment:
`#[cfg_attr(test, derive(Debug))]` prevents unit tests of crates that depend on this crate from using the `Debug` implementation, which I don't think you intended.

Either make `derive(Debug)` unconditional, or add a cargo feature and make the `derive(Debug)` depend on that feature.